### PR TITLE
Add Automatic pre-commit configuration for devShells.default & Migrate to nixfmt-classic

### PR DESCRIPTION
**Pull request by DrymarchonShaun**
This _originally_ started simply just to suggest a switch over to [nixfmt](https://github.com/NixOS/nixfmt), since the [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) repo was archived a few days ago (Jul 24, 2024) 

It was part way through that I realized this repo was set up to use `pre-commit` and that it was using a nixpkgs-fmt hook, so I started looking to see if nixfmt had its own, ended up seeing you mention forgetting to install pre-commit in #1 and then thinking _I wonder if theres a nix way to manage that, and hook it into the devshell to automate it some way_. So anyways, a couple hours later this is the result so far. It's using [cachix/git-hooks.nix](https://github.com/cachix/git-hooks.nix) to configure `pre-commit`, along with hooking into the default devshell to auto-create .pre-commit-config.yaml, and install the git hooks automatically (using either direnv, `nix develop` or nix-shell).

### CODE REVIEW

**Comment by DrymarchonShaun** [comment]
@EmergentMind No idea if you'd have any interest in this, I can make it a proper PR if you do. I've tested things somewhat, `pre-commit` is new to me, so I don't know if it's working entirely as it should, but it seemed to be? 

I can't 100% test to see if the formatting outright *broke* anything because the build fails without access to your nix-secrets repo, however I will say I've done things the same way on my fork and the build is working fine.

_No hard feelings if you aren't interested, jumping down this rabbit hole was a fun way to learn way more about Nix regardless. Out of all the Nix config repo's this one has been the easiest to understand and work with, and has worked great as a starting place for my own proper config._

**Comment by EmergentMind** [comment]
Hey, thanks for this!
I'm definitely interested in reviewing and merging this but will take a bit to look at it. I've been slammed with other priorities lately so sadly haven't been spending much time on my config. 
Really appreciate you taking the time.

File: flake.nix [hunk 1]
```diff
@@ -117,9 +117,8 @@
         import ./checks { inherit inputs system pkgs; }
       );
 
-      # TODO change this to something that has better looking output rules
-      # Nix formatter available through 'nix fmt' https://nix-community.github.io/nixpkgs-fmt
-      formatter = forAllSystems (system: nixpkgs.legacyPackages.${system}.nixfmt-classic);
+      # Nix formatter available through 'nixfmt' https://nix-community.github.io/nixpkgs-fmt
+      formatter = forAllSystems (system: nixpkgs.legacyPackages.${system}.nixfmt);
```
Comment (DrymarchonShaun) [inline flake.nix]: I believe it is currently not recommended to use the `nixfmt` package while we're migrating to the new rfc based nixfmt. Currently `nixfmt` *is* the same as `nixfmt-classic` but may switch to `nixfmt-rfc-style` at some point, with little warning. I wasn't sure which of the two you would rather use, so I went with `nixfmt-classic` as it is more similar to `nixpkgs-fmt`.
If I know which to use, I can clean up this PR to use one or the other from the beginning.

Comment (EmergentMind) [reply flake.nix]: Okay, good to know. Let's go with nixfmt-rfc-style then since it will be the way going forward.

**Review by EmergentMind — APPROVED** [review]

**Comment by EmergentMind** [comment]
Thanks for this. Greatly appreciated.

**Pull request merged**